packages/ui/src/components/PropertiesModal/PropertiesModal.tsx

```diff
@@ -19,16 +20,23 @@ interface IPropertiesModalProps {
 }
 
 export const PropertiesModal: FunctionComponent<IPropertiesModalProps> = (props) => {
+  const catalogService = useContext(CatalogContext);
   const tabs = useMemo(() => {
     switch (props.tile.type) {
       case CatalogKind.Component: {
-        return transformCamelComponentIntoTab(props.tile.rawObject as ICamelComponentDefinition);
+        return transformCamelComponentIntoTab(
+          catalogService.getComponent(CatalogKind.Component, props.tile.name) as ICamelComponentDefinition,
```

Comment:
Are these castings still needed? As far as I remember, the `getComponent` service returns the right type directly, based on the `CatalogKind` enum

Reply:
I checked. The thing is that the service can return the component definition or `undefined`, making the `transformCamelComponentIntoTab` to fail. I think we could update those functions to return an empty array  in case the tile definition is `undefined`, as we're happily accessing the following property:

```tsx
  const finalTabs: IPropertiesTab[] = [];

  let tab = transformPropertiesIntoTab(
    [
      {
        transformFunction: camelComponentPropertiesToTable,
        objectToTransform: componentDef.componentProperties, // Here
      },
    ],
    'Component Options',
  );
```